brats/transformations.py

```diff
@@ -200,3 +203,46 @@ def __call__(self, tensor: torch.Tensor) -> torch.Tensor:
         stds = tensor.std(dim=[1, 2, 3], keepdims=True)
         transformed = (tensor - means) / stds
         return transformed
+
+
+class OneHotEncoding:
+    """
+    From multiclass label image with one channel it creates N channel image,
+    where each channel is a mask of different class.
+    """
+
+    def __init__(self, classes: typing.List[int]):
+        self.classes = classes
+
+    def __call__(self, img: typing.Union[np.ndarray, torch.Tensor], ) -> typing.Union[np.ndarray, torch.Tensor]:
+        assert img.ndim == 4, "Tensor should have 4 dimensions (C,D,H,W)"
+        transformed = one_hot_encoding(img, self.classes)
+        return transformed
+
+
+@singledispatch
+def one_hot_encoding(img: typing.Any, classes: typing.List[int]):
+    raise TypeError("Img should be either np.ndarray of torch.Tensor")
+
+
+@one_hot_encoding.register(np.ndarray)
+def _(img: np.ndarray, classes: typing.List[int]) -> np.ndarray:
+    if 0 in classes:
+        classes.remove(0)  # Without the background label
+    new_shape = [len(classes)] + list(img.shape[1:])
+    transformed = np.zeros(new_shape)
+    for class_id, label in enumerate(filter(lambda label: label != 0, classes)):
+        transformed[class_id][img[0, ...] == label] = 1
```

Comment:
This 0 is very confusing to me, what does it represent?

Reply:
OK, this is connected to the indexing, I need to create a tensor with [N, D, H, W] shape, where N is number classes. So I do it and then want to fill it with the mask from the original one. 

If I would do
```python

transformed[class_id][mask == label] = 1

```
or
```python

transformed[mask == label] = 1

```
I would get index error, because there is a mismatch in shape. 
In first version, there would be too much indices in `mask` to properly index `transformed`.
In the second one, there would be correct number of indices, however we would try to index transformed (shape `[3, 155, 240, 240]`) with mask (shape `[1, 155, 240, 240]`). 

So what I do is to index each channel of `transformed` (`transformed[class_id]`) separately by squeezed version of `mask` (`mask[0, ...]`).

If you have a better solution, I would love to change that part of code as I really needed to focus in order to remember what I did here.

Reply:
Oh, I thought that the input parameter `img` has only three dimensions (D, H, W), but we add an additional one in the previous transformations. This was necessary in the case of binary classification, but do we need to keep that transformation which adds an aditional dimension in the case of multilcass, having this transformation? I guess yes, if we ever would want to go back to binary classification smoothly.

Reply:
In case of od DL, I prefer to keep the channel dimension - I believe it is a way to emphasize that this channel is there. Of course, additionally going back to binary is convenient this way.